gen/gen.go

```diff
@@ -94,50 +113,55 @@ func Struct(s string, dst io.Writer, gc Config) error {
 	return err
 }
 
-func generateFrom(schema avro.Schema, acc *data) string {
+func (g generator) generateFrom(schema avro.Schema, acc *data) string {
 	switch t := schema.(type) {
 	case *avro.RecordSchema:
 		typeName := strcase.ToCamel(t.Name())
 		fields := make([]field, len(t.Fields()))
 		for i, f := range t.Fields() {
 			fSchema := f.Type()
 			fieldName := strcase.ToCamel(f.Name())
-			typ := resolveType(fSchema, f.Prop("logicalType"), acc)
+			typ := g.resolveType(fSchema, acc)
 			tag := f.Name()
-			fields[i] = newField(fieldName, typ, tag)
+			fields[i] = g.newField(fieldName, typ, tag)
 		}
 		acc.Typedefs = append(acc.Typedefs, newType(typeName, fields))
 		return typeName
 	default:
-		return resolveType(schema, nil, acc)
+		return g.resolveType(schema, acc)
 	}
 }
 
-func resolveType(fieldSchema avro.Schema, logicalType interface{}, acc *data) string {
+func (g generator) resolveType(fieldSchema avro.Schema, acc *data) string {
 	var typ string
 	switch s := fieldSchema.(type) {
 	case *avro.RefSchema:
 		typ = resolveRefSchema(s)
 	case *avro.RecordSchema:
-		typ = generateFrom(s, acc)
+		typ = g.generateFrom(s, acc)
 	case *avro.PrimitiveSchema:
-		typ = resolvePrimitiveLogicalType(logicalType, typ, s)
-		if strings.Contains(typ, "time") {
-			addImport(acc, "time")
-		}
-		if strings.Contains(typ, "big") {
-			addImport(acc, "math/big")
+		logicalSchema := s.Logical()
+		if logicalSchema != nil {
+			typ = resolveLogicalSchema(logicalSchema.Type())
+			if strings.Contains(typ, "time") {
+				addImport(acc, "time")
+			}
+			if strings.Contains(typ, "big") {
+				addImport(acc, "math/big")
+			}
+		} else {
+			typ = primitiveMappings[s.Type()]
 		}
 	case *avro.ArraySchema:
-		typ = fmt.Sprintf("[]%s", generateFrom(s.Items(), acc))
+		typ = fmt.Sprintf("[]%s", g.generateFrom(s.Items(), acc))
 	case *avro.EnumSchema:
 		typ = "string"
 	case *avro.FixedSchema:
 		typ = fmt.Sprintf("[%d]byte", +s.Size())
```

Comment:
Fixed can also have a logical type.

Reply:
Do you have a sample schema in mind I can use to exercise that path?

Reply:
`{"type": "fixed", "name":"test", "size": 12, "logicalType": "duration"}`